libs/vkd3d/device.c

```diff
@@ -9528,61 +9528,68 @@ bool d3d12_device_validate_shader_meta(struct d3d12_device *device, const struct
-                device->vk_physical_device, &count, props)) != VK_SUCCESS)
-        {
-            ERR("Failed to query cooperative matrix properties.\n");
-            vkd3d_free(props);
-            return false;
-        }
+            if (VK_CALL(vkGetPhysicalDeviceCooperativeMatrixPropertiesKHR(
+                    device->vk_physical_device, &count, props)) != VK_SUCCESS)
+            {
+                ERR("Failed to query cooperative matrix properties.\n");
+                vkd3d_free(props);
+                return false;
+            }
 
-        for (i = 0; i < count; i++)
-        {
-            const VkCooperativeMatrixPropertiesKHR *fmt = &props[i];
+            for (i = 0; i < count; i++)
+            {
+                const VkCooperativeMatrixPropertiesKHR *fmt = &props[i];
 
-            if (fmt->AType == VK_COMPONENT_TYPE_UINT8_KHR)
-                supports_u8_a = true;
-            if (fmt->BType == VK_COMPONENT_TYPE_UINT8_KHR)
-                supports_u8_b = true;
+                if (fmt->AType == VK_COMPONENT_TYPE_UINT8_KHR)
+                    supports_u8_a = true;
+                if (fmt->BType == VK_COMPONENT_TYPE_UINT8_KHR)
+                    supports_u8_b = true;
 
-            if (fmt->KSize != 16 || fmt->MSize != 16 || fmt->NSize != 16 || fmt->scope != VK_SCOPE_SUBGROUP_KHR)
-                continue;
+                if (fmt->KSize != 16 || fmt->MSize != 16 || fmt->NSize != 16 || fmt->scope != VK_SCOPE_SUBGROUP_KHR)
+                    continue;
+
+                if (fmt->AType == VK_COMPONENT_TYPE_FLOAT16_KHR && fmt->BType == VK_COMPONENT_TYPE_FLOAT16_KHR)
+                {
+                    if (fmt->CType == VK_COMPONENT_TYPE_FLOAT32_KHR)
+                        supports_f32_16x16x16_f16 = true;
+                    else if (fmt->CType == VK_COMPONENT_TYPE_FLOAT16_KHR)
+                        supports_f16_16x16x16_f16 = true;
+                }
+            }
 
-            if (fmt->AType == VK_COMPONENT_TYPE_FLOAT16_KHR && fmt->BType == VK_COMPONENT_TYPE_FLOAT16_KHR)
+            vkd3d_free(props);
+            if (!supports_f16_16x16x16_f16 || !supports_f32_16x16x16_f16 || !supports_u8_a || !supports_u8_b)
             {
-                if (fmt->CType == VK_COMPONENT_TYPE_FLOAT32_KHR)
-                    supports_f32_16x16x16_f16 = true;
-                else if (fmt->CType == VK_COMPONENT_TYPE_FLOAT16_KHR)
-                    supports_f16_16x16x16_f16 = true;
+                ERR("Missing sufficient features to expose WMMA.\n");
+                return false;
             }
         }
-
-        vkd3d_free(props);
-        if (!supports_f16_16x16x16_f16 || !supports_f32_16x16x16_f16 || !supports_u8_a || !supports_u8_b)
+        else
         {
-            ERR("Missing sufficient features to expose WMMA.\n");
-            return false;
+            INFO("Allowing cooperative matrices to go through despite not being supported. RDoc workaround for now.\n");
```

Comment:
Seems a bit scary given that our own tests currently have no way to query for support and rely on pipeline compiles to fail; this is probably going to cause crashes on e.g. rdna2?

We have a renderdoc check somewhere to enable `HOST_CACHED` and friends already, could perhaps reuse that for now for the time being.

Reply:
Given how I'm done debugging, I can probably just get rid of that commit.